src/Native/System.Native/pal_io.cpp

```diff
@@ -359,6 +359,39 @@ int32_t FcntlSetPipeSz(int32_t fd, int32_t size)
 }
 
 extern "C"
+int32_t FcntlGetIsNonBlocking(int32_t fd)
+{
+    int flags = fcntl(fd, F_GETFL);
+    if (flags == -1)
+    {
+        return -1;
+    }
+
+    return (flags & O_NONBLOCK) == O_NONBLOCK ? 1 : 0;
+}
+
+extern "C"
+int32_t FcntlSetIsNonBlocking(int32_t fd, int32_t isNonBlocking)
+{
+    int flags = fcntl(fd, F_GETFL);
+    if (flags == -1)
+    {
+        return -1;
+    }
+
+    if (isNonBlocking == 0)
+    {
+        flags &= ~O_NONBLOCK;
+    }
+    else
+    {
+        flags |= O_NONBLOCK;
+    }
```

Comment:
It should not be common--each socket tracks its blocking mode--but the change is straightforward. Fixed.